WalletWasabi/Extensions/TaskExtensions.cs

```diff
@@ -78,12 +78,17 @@ public static async Task WithAwaitCancellationAsync(this Task me, TimeSpan timeo
 			=> await me.WithAwaitCancellationAsync(ThreadingHelpers.Cancelled, (int)timeout.TotalMilliseconds).ConfigureAwait(false);
 
 		public static async Task<TResult> RunAsScheduledAsync<TResult>(this Task<TResult> task, DateTimeOffset when, CancellationToken cancellationToken)
+		{
+			await WaitUntilAsync(when, cancellationToken).ConfigureAwait(false);
+			return await task.ConfigureAwait(false);
+		}
+
```

Comment:
How about adding this? 

```
		public static async Task RunAsScheduledAsync(this Task task, DateTimeOffset when, CancellationToken cancellationToken)
		{
			await WaitUntilAsync(when, cancellationToken).ConfigureAwait(false);
			await task.ConfigureAwait(false);
		}
```

Reply:
That's how it was before. We can create the `Unit` type or duplicate the code (not only here, also in `WabiSabiHttpApiClient`). I am okay with both but given we are using the `System.React.Unit` type everywhere then we could consider using the same concept here.